src/commands/Animal/dog.ts

```diff
@@ -16,23 +16,23 @@ import { KlasaMessage } from 'klasa';
 export default class extends SkyraCommand {
 
 	public async run(message: KlasaMessage) {
-		const embed = new MessageEmbed()
-			.setColor(await DbSet.fetchColor(message))
-			.setTimestamp();
+		const [color, image] = await Promise.all([
+			DbSet.fetchColor(message),
+			this.fetchImage()
+		]);
 
-		try {
-			const randomDogData = await fetch<DogResultOk>('https://dog.ceo/api/breeds/image/random', FetchResultTypes.JSON);
-			if (randomDogData && randomDogData.status === 'success') {
-				// Just in case the image is not a valid image url then fallthrough to the catch
-				if (!IMAGE_EXTENSION.test(randomDogData.message)) throw '💥';
-
-				embed.setImage(randomDogData.message);
-			}
-		} catch {
-			embed.setImage('https://i.imgur.com/cF0XUF5.jpg');
-		}
+		return message.sendEmbed(new MessageEmbed()
+			.setColor(color)
+			.setImage(image)
+			.setTimestamp());
+	}
 
-		return message.sendEmbed(embed);
+	private async fetchImage() {
+		const randomDogData = await fetch<DogResultOk>('https://dog.ceo/api/breeds/image/random', FetchResultTypes.JSON)
+			.catch(() => null);
+		return randomDogData?.status === 'success' && isImageURL(randomDogData.message)
+			? randomDogData.message
+			: 'https://i.imgur.com/cF0XUF5.jpg';
```

Comment:
Should make this a try-catch, to match commands like anime (where you made that change from this structure in the typorm branch)

Reply:
It's intentional to use `.catch(() => null)` because we had a `if (randomDogData && randomDogData.status === 'success'` before, in case `fetch` returned unexpected data (happened before tbh).

Making the catch like that allows me to handle both cases in the same if, thus making it much clearer.

Reply:
ah, fair enough.